Ability to enable/disable HVAC, Fan, Swing modes

lbfrias opened the pull request:
### Context

My Kolin AC units use gree protocols but do not support vertical fan swings, heating, and some other modes included by default in the integration so I wanted to create a way to disable modes that aren't supported by my AC.

Apologies for the documentation refactor but I was having a hard time reading the previous YAML example, so I'm not sure if the change I made for the README is okay or not, but let me know.

### PR description

- Added option to select which modes are enabled for HVAC, fan, and fan swings
- Removed translation helper and offloaded icon translation and localization to HASS translation key
- Added icons.json for icon translation for fan and swing modes
- Updated README.md; added table to list down parameters

### Screenshots

Some screenshots for the icon translation:

![image](https://github.com/user-attachments/assets/a687909d-2d3f-4a60-a3f1-995e0a5b3b00)
![image](https://github.com/user-attachments/assets/87d161f7-fc2a-428c-8caf-4362bcdcb947)

Some screenshots for the localization (Changing language in the user settings uses the language json files in the translation directory)
![image](https://github.com/user-attachments/assets/e8274898-e341-479f-b718-ba36f2df040a)
![image](https://github.com/user-attachments/assets/b2b83fc5-fb46-41ef-85b7-091c0da64526)

RobHofmann commented:
I might be wrong, but I think the translation system is not in your version, right? I see the translation helper getting deleted & also some properties/vars getting removed regarding languages.

Let me know.

RobHofmann commented:
Does this also mean we can get rid of the translation files? we now made those obsolete too, right?

lbfrias commented:
Note that there are some breaking changes in the climate.yaml. Can I see your climate.yaml? Looks like the CONF_HVAC_MODES is empty.

lbfrias commented:
Ahhh I see, I was not familiar with config flow through the UI so I was not able to test that one. I will fix this and try to push changes. Thanks for testing and letting me know!

lbfrias commented:
Hi, @RobHofmann.

I did some changes to fix the config flow and added the option to select and add custom HVAC, fan, and swing modes in both the UI and climate.yaml.

I also updated the README.md and made changes to the translation files for them to be applied in the config flow UI as well.

### Screenshots:

HVAC, fan, swing mode selectors:
![image](https://github.com/user-attachments/assets/d924f962-c0e5-45c9-be4e-c1b2086b1425)
![image](https://github.com/user-attachments/assets/03c962f7-86cc-4191-9056-c21c2ea0b34c)

Adding custom modes:
![image](https://github.com/user-attachments/assets/45238d3c-e297-411b-96ba-76403f82613e)


Translated selectors and options:
![image](https://github.com/user-attachments/assets/957cf1d7-c9e1-48a4-a375-71f6b1a50bd3)
![image](https://github.com/user-attachments/assets/2d6b2bca-c6e6-493c-b051-bd64f3de440c)

Can you please test and let me know if there's any issue? Thanks!

RobHofmann commented:
Tested it! works perfect man! Thanks a lot! Great release!
